zerver/views/message_fetch.py

```diff
@@ -1049,6 +1099,13 @@ def get_messages_backend(
         log_data["extra"] = "[{}]".format(",".join(verbose_operators))
 
     with get_sqlalchemy_connection() as sa_conn:
+        if anchor_val == "first_after_date":
+            assert target_date is not None
+            anchor = find_first_after_date_anchor(sa_conn, user_profile, realm, narrow, target_date)
+
+            if anchor == -1:
+                raise NoMessageFoundAfterDate(target_date)
```

Comment:
sorry if this is a dumb question (perhaps add a comment?) - why is no-message-found an error/exception and not just a normal behavior, i.e. returns zero messages?

put another way, I would have expected the API to return <up to n> messages between two time ranges, and zero messages to be an normal result.

Reply:
We were thinking specifically in terms of this feature of moving messages by recency when writing this. See messages around https://chat.zulip.org/#narrow/stream/378-api-design/topic/GET.20.2Fmessages.2F.7Bmsg_id.7D.20anchor.3Dfirst_after_date/near/1354215 for previous discussion

Feel free to add your thoughts there too, to encourage some discussion over this.